src/scripts/outlet-registration/index.js

```diff
@@ -0,0 +1,582 @@
+    if (err) {
+        return console.log(err);
+    }
+    var autoindicatorsLoader = new OutletRegistrator(JSON.parse(conf));
+    autoindicatorsLoader.loadLastEvents();
+});
+
+
+/**
+ * Class in charge of loading autoindicators turning them into datavalues.
+ */
+function OutletRegistrator(conf) {
+		
+    //used endpoints
+    this.endpoints = {
+        
+        EVENTS: "/events.json?orgUnit=[ROOT]&ouMode=DESCENDANTS&program=[PROGRAM]&startDate=",
+        EDITEVENTS: "/events/[EVENT]",
+        ORGUNITS: "/organisationUnits/[PARENT].json?includeChildren=true",
+        DATAVALUESETS: "/dataValueSets",
+        ORGUNIT: "/organisationUnits/",
+        ORGUNITGROUPORGUNIT: "/organisationUnitGroups/[OUGROUP]/organisationUnits/[ORGUNIT]",
+        ORGUNITDATASET: "/organisationUnits/[ORGUNIT]/dataSets/[DATASET]",
+        ORGUNITPROGRAM: "/organisationUnits/[ORGUNIT]/programs/[PROGRAM]",
+        OUTLETTYPE: "/organisationUnitGroups?filter=name:eq:[OUTLETTYPE]",
+        USERSFILTER: "/userCredentials?fields=id,username,userInfo&filter=username:eq:[USERNAME]",
+        USERORGUNITS: "/users/[USER]/organisationUnits/[ORGUNIT]"
+    };
+    
+    //This is the prefix of the Orgunit group names for Outlet Type
+    this.outletTypePrefix = "MM Type - ";
+    //This is the prefix for Myanmar
+    this.myanmarPrefix = "MM_"
+    //Count of imported org. units
+    this.orgUnitsCreated = 0;
+    //rest config
+    this.conf = conf;
+
+    //common auth, endpoint config
+    this.requestOptions = {
+        headers: {
+            authorization: 'Basic ' + this.conf.auth,
+        },
+        url: this.conf.protocol + "://" + this.conf.url + ((this.conf.apiVersion == "")?"":"/" +this.conf.apiVersion)
+    }
+    console.log("\nConfig:\n", JSON.stringify(this.conf, null, "\t"));
+};
+
+/**
+ * Returns an object with url and auth info for the given endpoint
+ * @param endpoint The endpoint with the params included
+ */
+OutletRegistrator.prototype.prepareOptions = function(endpoint) {
+    var options = Object.assign({}, this.requestOptions);
+    options.url += endpoint;
+    return options;
+}
+
+/**
+ * Loads every event from the last 30 days ()
+ */
+OutletRegistrator.prototype.loadLastEvents = function() {
+    console.log("\nLoading events...");
+    var _this = this;
+    //Ask for 'Events'
+    var requestData = this.prepareOptions(this.endpoints.EVENTS);
+    requestData = this.prepareEventsRequest(requestData);
+    request(requestData, function(error, response, body) {
+        _this.events = JSON.parse(body).events;
+        console.log("Found " + _this.events.length + " events");
+        _this.events = _this.events.filter(event => _this.isAlreadyImported(event)==false);
+        console.log("Org. Units to import ", _this.events.length);
+                
+        if (_this.events.length == 0) {
+        	console.log ("No org. units to import");
+        	return;
+        }
+        
+        _this.events.sort(function(a,b){
```

Comment:
Can we change the variable name to something more clear? Maybe event1 and event2? Don't know...